components/brave_wallet/browser/solana_provider_impl.h

```diff
@@ -151,12 +156,18 @@ class SolanaProviderImpl final : public mojom::SolanaProvider,
   void OnTransactionStatusChanged(mojom::TransactionInfoPtr tx_info) override;
   void OnTxServiceReset() override {}
 
+  // content_settings::Observer:
+  void OnContentSettingChanged(const ContentSettingsPattern& primary_pattern,
+                               const ContentSettingsPattern& secondary_pattern,
+                               ContentSettingsType content_type) override;
+
   base::flat_map<std::string, SignAndSendTransactionCallback>
       sign_and_send_tx_callbacks_;
   // Pending callback and arg are for waiting user unlock before connect
   ConnectCallback pending_connect_callback_;
   absl::optional<base::Value::Dict> pending_connect_arg_;
 
+  const raw_ref<HostContentSettingsMap> host_content_settings_map_;
```

Comment:
nit: through chromium's code this is typically captured by raw_ptr

Reply:
use `raw_ref` here is intentional to enforce that `host_content_settings_map_` would never be null. 
Same usage is in `brave/components/permissions/permission_lifetime_manager.h` written by @goodov